core/src/map.cpp

```diff
@@ -216,30 +224,28 @@ void Map::loadSceneAsync(const char* _scenePath, bool _useScenePosition,
         nextScene = impl->nextScene;
     }
 
-    runAsyncTask([nextScene, _onMapReady, _onMapReadyUserData, _sceneUpdates, _onSceneUpdateError, this](){
+    runAsyncTask([nextScene, _onSceneReady, _sceneUpdates, this](){
 
-            bool newSceneLoaded = SceneLoader::loadScene(platform, nextScene, _sceneUpdates, _onSceneUpdateError);
+            bool newSceneLoaded = SceneLoader::loadScene(platform, nextScene, _sceneUpdates);
 
-            impl->jobQueue.add([nextScene, newSceneLoaded, _onMapReady, _onMapReadyUserData, _onSceneUpdateError, this]() {
+            impl->jobQueue.add([nextScene, newSceneLoaded, _onSceneReady, this]() {
                     {
                         std::lock_guard<std::mutex> lock(impl->sceneMutex);
                         if (nextScene == impl->nextScene) {
                             impl->nextScene.reset();
                         } else {
                             // loadScene[Async] was called in the meantime.
+                            if (_onSceneReady) { _onSceneReady(false, {}); }
                             return;
                         }
                     }
 
                     if (newSceneLoaded) {
                         auto s = nextScene;
                         impl->setScene(s);
-                        applySceneUpdates(_onSceneUpdateError);
-
-                        if (_onMapReady) {
-                            _onMapReady(_onMapReadyUserData);
-                        }
                     }
+
+                    if (_onSceneReady) { _onSceneReady(newSceneLoaded, {}); }
```

Comment:
Or the aim is for the user to use the `status` of the loadScene/sceneUpdate and do accordingly.